File: tests/sdk/nodejs/test/eventbridge.test.mjs
import { test } from "node:test";
import assert from "node:assert/strict";
import {
  CreateEventBusCommand,
  DeleteEventBusCommand,
  DeleteRuleCommand,
  DescribeRuleCommand,
  PutRuleCommand,
} from "@aws-sdk/client-eventbridge";
import { newEvents, requireReady, uniquePrefix } from "../lib/helpers.mjs";

test("EventBridge bus and rule round-trip", async (t) => {
  if (!(await requireReady(t))) return;
  const client = newEvents();
  const prefix = uniquePrefix();
  const busName = `${prefix}-bus`;
  const ruleName = `${prefix}-rule`;

  await client.send(new CreateEventBusCommand({ Name: busName }));
  t.after(async () => {
    try {
      await client.send(new DeleteEventBusCommand({ Name: busName }));
    } catch {
      /* ignore */
    }
  });

  await client.send(
    new PutRuleCommand({
      Name: ruleName,
      EventBusName: busName,
      EventPattern: '{"source":["noctaxris.sdk"]}',
      State: "ENABLED",
    }),
  );
  t.after(async () => {
    try {
      await client.send(
        new DeleteRuleCommand({ Name: ruleName, EventBusName: busName }),
      );
    } catch {
      /* ignore */
    }
  });

  const desc = await client.send(
    new DescribeRuleCommand({ Name: ruleName, EventBusName: busName }),
  );
  assert.equal(desc.Name, ruleName);

  await client.send(
    new DeleteRuleCommand({ Name: ruleName, EventBusName: busName }),
  );
  await client.send(new DeleteEventBusCommand({ Name: busName }));
});

File: tests/sdk/go/eventbridge_test.go
package sdk_test

import (
	"context"
	"testing"

	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/service/eventbridge"
	"github.com/aws/aws-sdk-go-v2/service/eventbridge/types"
)

func TestEventBridgeBusAndRuleRoundTrip(t *testing.T) {
	requireReady(t)
	cfg := loadAWSConfig(t)
	client := newEvents(t, cfg)
	ctx := context.Background()
	prefix := uniquePrefix(t)

	busName := prefix + "-bus"
	_, err := client.CreateEventBus(ctx, &eventbridge.CreateEventBusInput{Name: aws.String(busName)})
	if err != nil {
		t.Fatalf("CreateEventBus: %v", err)
	}
	t.Cleanup(func() {
		_, _ = client.DeleteEventBus(ctx, &eventbridge.DeleteEventBusInput{Name: aws.String(busName)})
	})

	ruleName := prefix + "-rule"
	_, err = client.PutRule(ctx, &eventbridge.PutRuleInput{
		Name:         aws.String(ruleName),
		EventBusName: aws.String(busName),
		EventPattern: aws.String(`{"source":["noctaxris.sdk"]}`),
		State:        types.RuleStateEnabled,
	})
	if err != nil {
		t.Fatalf("PutRule: %v", err)
	}
	t.Cleanup(func() {
		_, _ = client.DeleteRule(ctx, &eventbridge.DeleteRuleInput{
			Name:         aws.String(ruleName),
			EventBusName: aws.String(busName),
		})
	})

	desc, err := client.DescribeRule(ctx, &eventbridge.DescribeRuleInput{
		Name:         aws.String(ruleName),
		EventBusName: aws.String(busName),
	})
	if err != nil {
		t.Fatalf("DescribeRule: %v", err)
	}
	if desc.Name == nil || *desc.Name != ruleName {
		t.Fatalf("DescribeRule unexpected: %+v", desc)
	}

	_, err = client.DeleteRule(ctx, &eventbridge.DeleteRuleInput{
		Name:         aws.String(ruleName),
		EventBusName: aws.String(busName),
	})
	if err != nil {
		t.Fatalf("DeleteRule: %v", err)
	}
	_, err = client.DeleteEventBus(ctx, &eventbridge.DeleteEventBusInput{Name: aws.String(busName)})
	if err != nil {
		t.Fatalf("DeleteEventBus: %v", err)
	}
}
